File: cpp/排序/插入排序.cpp
//插入排序
//时间复杂度 O(N*2)

#include <iostream>
using namespace std;

void swap(int &a,int &b)
{
	int temp = a;
	a = b;
	b = temp;
}

void InsertSort(int *arr,int n)
{
	//i 为有序数组的边界 
	for(int i = 1; i < n; i++)
	{
		for(int j = i-1; j >= 0; i--)
		{
			if(arr[j] > arr[j+1])
				swap(arr[j],arr[j+1]);
			else
				break;
		}
	}
} 

int main()
{
	int n;
	cin >> n;
	int *arr = new int[n];
	for(int i = 0; i < n; i++)	
		cin >> arr[i];
	InsertSort(arr,n);
	for(int i = 0; i < n; i++)	
		cout << arr[i] << " " ;
	cout << endl;
	system("pause");
	return 0;
}

File: py/排序/插入排序.py
# 插入排序
class InsertSort:
    def __init__(self, arr):
        self.arr = arr
        self.n = len(arr)

    def _swap(self, i, j):
        self.arr[i], self.arr[j] = self.arr[j], self.arr[i]
    
    def _insert_sort(self):
        for i in range(1, self.n):
            for j in range(i-1, -1, -1):
                if self.arr[j] > self.arr[j + 1]:
                    self._swap(j, j + 1)
                else:
                    break
    
    def get_sorted_arr(self):
        self._insert_sort()
        return self.arr
    
if __name__=='__main__':
    arr = list(map(int, input("请输入待排序数组：\n").split()))
    insert_sort = InsertSort(arr)
    print("插入排序结果：\n", insert_sort.get_sorted_arr())